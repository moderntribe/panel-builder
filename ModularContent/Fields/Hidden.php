<?php


namespace ModularContent\Fields;
use ModularContent\Panel;

/**
 * Class Hidden
 *
 * @package ModularContent\Fields
 *
 * A hidden field. It will be in the HTML, but not visible to the user.
 */
class Hidden extends Field {
	/**
	 * Massage submitted data before it's saved.
	 *
	 * @param mixed $data
	 * @return string
	 */
	public function prepare_data_for_save( $data ) {
		return (string) parent::prepare_data_for_save( $data );
	}
}