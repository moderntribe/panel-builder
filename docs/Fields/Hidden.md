## Hidden

A hidden field. It will be in the HTML, but not visible to the user.

### Parameters

||||
|---|---|---|
| `name`        | `string` | The unique name for this field                 |

### Example

```php
$field = new Hidden( array(
  'name' => 'secret',
) );
```