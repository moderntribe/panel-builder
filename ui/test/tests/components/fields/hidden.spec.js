/* global expect b:true */

import React from 'react';
import { shallow } from 'enzyme';
import Hidden from 'components/fields/hidden';
import renderer from 'react-test-renderer';

describe('Hidden field', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(<Hidden />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders as a <div>', () => {
		const wrapper = shallow(<Hidden />);
		expect(wrapper.type()).toEqual('div');
	});

	it('renders a hidden input', () => {
		const wrapper = shallow(<Hidden />);
		expect(wrapper.find('input[type="hidden"]').length).toEqual(1);
	});

	it('allows you to set properties', () => {
		const wrapper = shallow(<Hidden name="test-name" default="test-value" />);
		expect(wrapper.find('input[type="hidden"]').length).toEqual(1);
		expect(wrapper.find('input[type="hidden"]').props().value).toEqual('test-value');
		expect(wrapper.find('input[type="hidden"]').props().name).toEqual('test-name');
	});
});
