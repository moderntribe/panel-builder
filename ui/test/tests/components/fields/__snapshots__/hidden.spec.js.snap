exports[`Hidden field renders correctly 1`] = `
<div
  className="panel-field">
  <input
    name=""
    type="hidden"
    value="" />
</div>
`;
